spec/lib/modules/ai_bot/tools/search_settings_spec.rb

```diff
@@ -5,13 +5,28 @@
   let(:bot_user) { DiscourseAi::AiBot::EntryPoint.find_user_from_model(llm_model.name) }
   let(:llm) { DiscourseAi::Completions::Llm.proxy("custom:#{llm_model.id}") }
 
+  let(:fake_settings) do
+    [
+      { setting: "default_locale", description: "The default locale for the site", plugin: "core" },
+      { setting: "min_post_length", description: "The minimum length of a post", plugin: "core" },
+      {
+        setting: "ai_bot_enabled",
+        description: "Enable or disable the AI bot",
+        plugin: "discourse-ai",
+      },
+      { setting: "min_first_post_length", description: "First post length", plugin: "core" },
+    ]
+  end
+
   before do
     SiteSetting.ai_bot_enabled = true
     toggle_enabled_bots(bots: [llm_model])
   end
 
-  def search_settings(query)
-    described_class.new({ query: query }, bot_user: bot_user, llm: llm)
+  def search_settings(query, mock: true)
+    search = described_class.new({ query: query }, bot_user: bot_user, llm: llm)
+    search.all_settings = fake_settings if mock
```

Comment:
I think you need an `after` block to restore the original value of this, or reset it to nil so the memoization can kick in and re-initialize the settings in [search_settings.rb](https://github.com/discourse/discourse-ai/pull/780/files#diff-2f1ab10bdb1709b638c2b05ff7f7c0940fc62db89332de4c429b76570f294f90)

Reply:
Otherwise some test flakiness may occur

Reply:
An alternative approach is to do `DiscourseAi::AiBot::Tools::SearchSettings.any_instance.stubs(:site_settings).returns(mock)`, then that will automatically get cleaned up

Reply:
Or doing the above with `SiteSetting.all_settings` would avoid needing to add an `all_settings` method in that class entirely :)

Reply:
no need, this is just defining an instance :) the instance and the override vanish after the test

Reply:
Oh wow duh :facepalm: